presto-main/src/main/java/com/facebook/presto/operator/simple/ProcessorInput.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.simple;
+
+import com.facebook.presto.operator.simple.SimpleOperator.ProcessorState;
+
+import javax.annotation.Nullable;
+
+public interface ProcessorInput<T>
+{
+    /**
+     * Add input to the processor. A null input indicates that no more
+     * input will be supplied.
+     *
+     * @return current state of the processor: {NEEDS_INPUT | HAS_OUTPUT}
+     */
+    ProcessorState addInput(@Nullable T input);
```

Comment:
Split into `addInput` and `noMoreInput`. Both return ProcessorState.

Rename `NEEDS_INPUT` to `NO_OUTPUT`, to indicate it's a state where it can take input or is done.